pages/en/lb4/Application.md

```diff
@@ -0,0 +1,223 @@
+    const app = this; // For clarity.
+    app.controller(SamoflangeController);
+    app.controller(DoohickeyController);
+  }
+
+  async start() {
+    // This is where you would asynchronously retrieve servers, providers and
+    // other components to configure them before launch.
+    const server = await app.getServer(RestServer);
+    server.bind('rest.port').to(8080);
+    server.api(WidgetApi);
+    return await super.start();
+  }
+
+  async stop() {
+    // This is where you would do whatever is necessary before stopping your
+    // app (graceful closing of connections, flushing buffers, etc)
+    console.log('Widget application is shutting down...')
+    await super.stop();
+  }
+}
+
+```
+
+## Configuring your application
+Your application can be configured with constructor arguments, bindings, or
+a combination of both.
+
+### Binding configuration
+Binding is the most commonly-demonstrated form of application configuration
+throughout our examples, and in most cases, is the recommended method for
+setting up your application.
+
+In addition to the binding functions provided by [Context](Context.html),
+the `Application` class also provides some sugar functions for commonly used
+bindings, like `component`, `server` and `controller`:
+
+```ts
+export class MyApplication extends Application {
+  constructor() {
+    super();
+  }
+  this.component(MagicSuite);
+  this.server(RestServer, 'public');
+  this.server(RestServer, 'private');
+
+  this.controller(FooController);
+  this.controller(BarController);
+  this.controller(BazController);
+}
+```
+
+You can find a complete list of these functions on the
+[`Application`](http://apidocs.loopback.io/@loopback%2fcore/#Application) API
+docs page.
+
+Additionally, you can use more advanced forms of binding to fine-tune your
+application's configuration:
+
+```ts
+export class MyApplication extends Application {
+  constructor() {
+    super();
+  }
+  this.server(RestServer);
+  this.controller(FooController);
+  this.bind('fooCorp.logger').toProvider(LogProvider);
+  this.bind('repositories.widget')
+    .toClass(WidgetRepository)
+    .inScope(BindingScope.SINGLETON);
+}
+```
+In the above example:
+- injection calls for `fooCorp.logger` will be handled by the `LogProvider`
+  class.
+- injection calls for `repositories.widget` will be handled by a singleton
+instance of the `WidgetRepository` class.
+
+### Constructor configuration
+
+The `Application` class constructor also accepts an
+[`ApplicationConfig`](http://apidocs.strongloop.com/@loopback%2fcore/#ApplicationConfig)
+object which contains three collections: `components`, `controllers` and `servers`.
+It will automatically create bindings for each of these collections as a part
+of defining the application instance.
+
+{% include note.html content="
+  More advanced binding configuration such as provider binding, or binding scopes
+  are not possible with the constructor-based configuration approach.
```

Comment:
In this configuration method, should we mention what the default binding scopes would be?